### PR TITLE
Fix isActive for CMS page with "/" as URL

### LINKED ISSUE #478: Menu item not set as active
### Winter CMS Build

dev-develop

### PHP Version

7.4

### Database engine

MySQL/MariaDB

### Plugins installed

Winter.staticPage

### Issue description

Winter.staticPage menu don't set item as active if his pointing to a 'CMS page' with `url = "/"` configuration.

### Steps to replicate

Add a CMS page with this configuration in the active theme : 
```
title = "Welcome"
url = "/"
layout = "default"
```

Add a reference to this home page in a Winter.staticPage menu : 
- Type of the menu item 'URL', url value '/' => menu item is maked as active
- Type of the menu item 'CMS page' referencing the Welcome page => menu item is NOT maked as active

### Workaround

_No response_

### PR DESCRIPTION
**Pull request by mjauvin**
Fix #478 

<a href="https://gitpod.io/#https://github.com/wintercms/winter/pull/479"></a>

### CODE REVIEW

File: modules/cms/classes/Page.php [hunk 1]
```diff
@@ -209,7 +209,7 @@ public static function resolveMenuItem($item, string $url, Theme $theme)
 
             $result = [];
             $result['url'] = $pageUrl;
-            $result['isActive'] = $pageUrl == $url;
+            $result['isActive'] = $pageUrl === trim($url, "/");
```
Comment (LukeTowers) [inline modules/cms/classes/Page.php]: Can you add some unit tests for this? And wouldn't it make more sense to rtrim() both $pageUrl and $url?

**Comment by mjauvin** [comment]
Added unit test @LukeTowers

File: tests/unit/cms/classes/PageTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use Cms\Classes\Page;
+use Cms\Classes\Theme;
+
+class PageTest extends TestCase
+{
+    public function setUp() : void
+    {
+        parent::setUp();
+    }
+
```
Comment (LukeTowers) [inline tests/unit/cms/classes/PageTest.php]:
```suggestion
```

File: tests/unit/cms/classes/PageTest.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            'reference' => 'index',
+        ];
+
+        // fake current URL to be the same as menu item reference page with extra "/" at the end.
+        $url = $controller->pageUrl($item->reference) . '/';
+
+        $result = Page::resolveMenuItem($item, $url, $theme);
+
+        $this->assertTrue($result['isActive']);
+    }
```
Comment (LukeTowers) [inline tests/unit/cms/classes/PageTest.php]:
```suggestion
    public function testResolveMenuItem()
    {
        $theme = Theme::load('test');
        $controller = new Controller;

        $item = (object) [
            'type' => 'cms-page',
            'reference' => 'index',
        ];

        // Check to make sure that resolved menuItems for the provided URL are considered active
        // with or without a trailing slash
        $url = $controller->pageUrl($item->reference);
        $trailingUrl = $url . '/';

        $result = Page::resolveMenuItem($item, $url, $theme);
        $this->assertTrue($result['isActive']);

        $result = Page::resolveMenuItem($item, $trailingUrl, $theme);
        $this->assertTrue($result['isActive']);
    }
```

File: modules/cms/classes/Page.php [hunk 4]
```diff
@@ -209,7 +209,7 @@ public static function resolveMenuItem($item, string $url, Theme $theme)
 
             $result = [];
             $result['url'] = $pageUrl;
-            $result['isActive'] = $pageUrl == $url;
+            $result['isActive'] = rtrim($pageUrl, "/") === rtrim($url, "/");
```
Comment (bennothommo) [inline modules/cms/classes/Page.php]:
```suggestion
            $result['isActive'] = rtrim($pageUrl, '/') === rtrim($url, '/');
```

**Pull request merged**